scripts/synthetic/generate_pipeline_results_files.py

```diff
@@ -160,53 +163,66 @@ def _generate_resultfiles_json(
     help="the sample metadata file",
 )
 @click.option(
-    "--output-csv-file",
+    "--output-results-csv-file",
     type=click.Path(dir_okay=False, writable=True),
-    required=True,
-    help="output file merging the results from ncov and pangolin",
+    default="results.csv",
+    help="CSV file containing the summary of results",
 )
 @click.option(
-    "--output-json-file",
+    "--output-results-json-file",
     type=click.Path(dir_okay=False, writable=True),
-    required=True,
-    help="output file containing all the expected files per sample",
+    default="results.json",
+    help="JSON file containing the summary of results",
+)
+@click.option(
+    "--output-resultfiles-json-file",
+    type=click.Path(dir_okay=False, writable=True),
+    default="resultfiles.json",
+    help="JSON file containing all the expected files per sample",
 )
 @click.option(
     "--output-path",
     type=str,
     required=True,
     help="output_path output path where sample result files are stored (e.g. s3://bucket/path/analysis_run)",
 )
-def generate_results(
+def generate_pipeline_results_files(
     metadata_file: str,
-    output_csv_file: str,
-    output_json_file: str,
+    output_results_csv_file: str,
+    output_results_json_file: str,
+    output_resultfiles_json_file: str,
     output_path: str,
 ) -> None:
     """
     Generate pipeline results files
     """
+    # generate a cycle list so that generated data for each sample is predictable
+    SYNTHETIC_DATA_POOL = cycle(SYNTHETIC_DATA)
+
     df_metadata = load_data_from_csv(Path(metadata_file), EXPECTED_METADATA_HEADERS)
-    all_samples = df_metadata[SAMPLE_ID].tolist()
+    all_samples = sorted(df_metadata[SAMPLE_ID].tolist())
 
     # generate some synthetic results.
-    successfull_samples = all_samples
+    successful_samples = all_samples
     qc_unrelated_failing_samples: List[str] = []
 
-    synthetic_sample_results = [random.choice(SYNTHETIC_DATA) for sample in successfull_samples]
+    synthetic_sample_results = [next(SYNTHETIC_DATA_POOL) for sample in successful_samples]
 
     synthetic_results = {
-        SAMPLE_ID: successfull_samples,
+        SAMPLE_ID: successful_samples,
         QC: [s[QC] for s in synthetic_sample_results],
         LINEAGE: [s[LINEAGE] for s in synthetic_sample_results],
         RD: [s[RD] for s in synthetic_sample_results],
     }
     df_synthetic = pd.DataFrame(data=synthetic_results)
 
-    _generate_results_csv(all_samples, df_synthetic, qc_unrelated_failing_samples, output_csv_file)
-    _generate_resultfiles_json(successfull_samples, output_path, Path(output_json_file))
+    _generate_results_csv(all_samples, df_synthetic, qc_unrelated_failing_samples, output_results_csv_file)
+
+    csv_to_json(output_results_csv_file, output_results_json_file, SAMPLE_ID)
```

Comment:
Add a results.json file too as for the sars-cov-2 pipeline so if we ever decide to switch from results.csv to results.json, the pipeline is already ready for this.